cibuildwheel/macos.py

```diff
@@ -182,6 +183,12 @@ def setup_python(python_configuration: PythonConfiguration,
 
 
 def build(options: BuildOptions) -> None:
+    if options.architectures != [Architecture.x86_64]:
+        raise ValueError(textwrap.dedent(f'''
+            Invalid archs option {options.architectures}. macOS only supports x86_64 for the moment.
+            If you want to set emulation architectures on Linux, use CIBW_ARCHS_LINUX instead.
```

Comment:
Just had a thought, should we update the CI configurations where setting the env var in a "global" level will make non-Linux builds in the matrix fail because of this assertion?

Reply:
I'm not sure what you mean, "the CI configurations"?

Reply:
(I think) I meant the documentation on "how to configure CI to use cibuildwheel" , so for example in the sample file below:

https://github.com/joerick/cibuildwheel/pull/482/files?short_path=2e04679#diff-fcf87e56f37482d09dcba260e0c81bf1817b22e1bd964091931002ea67918fabR42

If I'm not mistaken, specifying the env var there without os restriction, will set `options.architerctures` for MacOS as well, which raises an error here on line 187 right ?

Reply:
Here I'm setting `CIBW_ARCHS_LINUX`, which won't affect the value of `options.architectures` on macOS/Windows.

Reply:
@asfaltboy means (I think) that there is no test that sets `CIBW_ARCHS` and verifies that an exception is thrown. I think it's something we could add later, better to get this in for now? Personally, I'm pretty bad about forgetting to add tests for expected failures, and it has bit me a couple of times.

Reply:
We have the option parsing unit tests, which verify what gets passed through to the build functions, so maybe that's covered? But you're right @asfaltboy , the exception isn't tested at all. Maybe I should add something.

> Personally, I'm pretty bad about forgetting to add tests for expected failures, and it has bit me a couple of times.

me too :)

Reply:
that test added in 5ee663f